src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form.ts

```diff
@@ -578,16 +659,6 @@ export function useForm<T extends FormData = FormData, I extends FormData = T>(
   // ----------------------------------
   // -- EFFECTS
   // ----------------------------------
-
-  useEffect(() => {
```

Comment:
That was a bug. We don't want the `defaultValue` internal ref to be updated when the object ref changes

```js
const { form } = useForm({
  defaultValue: {
    ... // this object will always be a new reference when the component re-renders
  }
});
```